### PR TITLE
Dev/policy object

### PR DESCRIPTION
**Pull request by jpkrajewski**
# Pull Request summary:
Coverage for policy objects
In vManage: **Configuration > Policy Groups > Group of Interest**
image
image

# Description of changes:

- Added GET, POST, PUT, DELETE methods for Group of Interest
- Mapped all models from Group of Interest
- Changed `Parcel` to parse data from GET endpoint
- Created ready to go demo script
**How to run a demo: `py .\examples\parcel_configuration_guide.py <ip address> <port> <username> <password>`**

- Created new folder with the models and added top level stub that gives access to them in:
catalystwan\models\configuration\feature_profile\sdwan\interest_groups\__init__.py
image

# Checklist:
- [X] Make sure to run pre-commit before committing changes
- [X] Make sure all checks have passed
- [X] PR description is clear and comprehensive
- [X] Mentioned the issue that this PR solves (if applicable)
- [X] Make sure you test the changes

### CODE REVIEW

File: catalystwan/endpoints/configuration/feature_profile/sdwan/policy_object.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
+# mypy: disable-error-code="empty-body"
+# flake8: noqa
```
Comment (sbasan) [inline catalystwan/endpoints/configuration/feature_profile/sdwan/policy_object.py:2]: please don't forget to remove that before merging to main.

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/application_list.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
+from enum import Enum
+from typing import List, Union
+
+from pydantic import BaseModel, Field, PrivateAttr
+
+from catalystwan.api.configuration_groups.parcel import Global, _ParcelBase
+from catalystwan.models.configuration.feature_profile.sdwan.policy_object.object_list_type import PolicyObjectListType
+
+
+class ApplicationType(str, Enum):
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/policy_object/application_list.py:10]: Looks like it is not statically defined list, the available options are provided by vManage:
`/dataservice/app-registry/applications`
Please remove that enum.

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
+from enum import Enum
+from typing import List
+
+from pydantic import BaseModel, PrivateAttr
+
+from catalystwan.api.configuration_groups.parcel import Global, _ParcelBase
+from catalystwan.models.configuration.feature_profile.sdwan.policy_object.object_list_type import PolicyObjectListType
+
+
+class ColorType(str, Enum):
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py:10]: already defined numerous times, lets reuse:
`catalystwan/models/common.py::TLOCColorEnum`

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+class ColorEntry(BaseModel):
+    color: Color
+
+
+class ColorData(BaseModel):
+    entries: List[ColorEntry]
+
+
+class ColorPayload(_ParcelBase):
+    _payload_endpoint: PolicyObjectListType = PrivateAttr(default=PolicyObjectListType.COLOR)
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py:48]: Coupling a model to endpoint as an attribute of model does not fit current approach.
This coupling should be done in `endpoint` definition.

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/ipv6_prefix_list.py [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+class Ipv6Address(Global):
+    value: IPv6Address
+
+
+class Ipv6PrefixLength(Global):
+    value: int
+
+
+class IPv6PrefixListEntry(BaseModel):
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/policy_object/ipv6_prefix_list.py:18]: Maybe add static method which will produce `IPv6PrefixListEntry` from `IPv6Network` = `IPv6Address` + prefixlen

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/prefix_list.py [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+class Ipv4Address(Global):
+    value: IPv4Address
+
+
+class Ipv4PrefixLength(Global):
+    value: int
+
+
+class PrefixListEntry(BaseModel):
```
Comment (sbasan) [inline catalystwan/models/configuration/feature_profile/sdwan/policy_object/prefix_list.py:18]: Maybe add static method which will produce `IPv4PrefixListEntry` from `IPv4Network` = `IPv4Address` + prefixlen

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/application_list.py (hunk 2 above)
Comment (jpkrajewski) [reply catalystwan/models/configuration/feature_profile/sdwan/policy_object/application_list.py]: deleted

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py (hunk 3 above)
Comment (jpkrajewski) [reply catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py]: deleted model and reused `catalystwan/models/common.py::TLOCColorEnum`

File: catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py (hunk 4 above)
Comment (jpkrajewski) [reply catalystwan/models/configuration/feature_profile/sdwan/policy_object/color_list.py:15]: will create mapping

File: catalystwan/endpoints/configuration/feature_profile/sdwan/policy_object.py (hunk 1 above)
Comment (jpkrajewski) [reply catalystwan/endpoints/configuration/feature_profile/sdwan/policy_object.py:2]: will delete

**Review by sbasan — DISMISSED** [review]
lgtm

**Pull request merged**